rust/src/ssh/ssh.rs

```diff
@@ -64,7 +65,8 @@ pub enum SSHEvent {
 }
 
 #[repr(u8)]
-#[derive(Copy, Clone, PartialOrd, PartialEq, Eq)]
+#[derive(AppLayerState, Copy, Clone, PartialOrd, PartialEq, Eq)]
+#[sc_alstate_strip_prefix = "SshState"]
```

Comment:
I get it, but this looks confusing..  Would it work to have a `suricata` attribute namespace?
```
#[suricata(strip_prefix = "SshState")]
```
could we then add other fields inside there even if only used by other proc macros?  Just thinking out loud here.

Reply:
I do not know rust `attribute namespace` ... Do you have some doc about it ?

`#[suricata(strip_prefix = "SshState")]` looks like an attribute `suricata` with a List cf https://docs.rs/syn/1.0.109/syn/enum.Meta.html where I use here a [NameValue](https://docs.rs/syn/1.0.109/syn/enum.Meta.html#namevalue)

I used `sc_` prefix here, but I can make it `suricata_alstate_strip_prefix`

I think it is more rust-idiomatic to have one helper attribute dedicated to the derive macro from https://doc.rust-lang.org/reference/procedural-macros.html#derive-macro-helper-attributes

Reply:
I was getting inspiration from `serde`, where you have proc macros `serde::Serialize` and `serde::Deserialize`, but then the attributes are all under `serde`, like: `#[serde(rename_all = "snake_case")]` or whatever. Seem very similar to this, where we have multiple proc macros, but put the attributes under the same name space.

Reply:
Will give it some more thought.

Where do you see rust doc about name space ?

Reply:
I haven't looked.  Just a serde user.

Reply:
Really I just think it looks cleaner for an API that will be around for a while. But it probably doesn't matter too much.

Reply:
I did this in https://github.com/OISF/suricata/pull/13050

> But it probably doesn't matter too much.

I agree

> Really I just think it looks cleaner for an API that will be around for a while

I do not agree : it does not have to stay for a while, we can change it if we want, right ?

And I do not think it is cleaner, because I do not think it is a namespace (but I may miss something here) but it is a "derive macro helper attribute" and I think these are meant by rust to be dedicated to one derive macro

So, I guess we may talk about it again when we want to use this `suricata` attribute in another context ;-)